wfe/web-front-end.go

```diff
@@ -135,12 +136,25 @@ func verifyPOST(request *http.Request) ([]byte, jose.JsonWebKey, error) {
 	// RA.  However the WFE is the RA's only view of the outside world
 	// *anyway*, so it could always lie about what key was used by faking
 	// the signature itself.
-	if err = jws.Verify(); err != nil {
-		return nil, zeroKey, err
+	if len(parsedJws.Signatures) > 1 {
+		wfe.log.Debug(fmt.Sprintf("Too many signatures on POST"))
+		return nil, nil, errors.New("Too many signatures on POST")
+	}
+	if len(parsedJws.Signatures) == 0 {
+		wfe.log.Debug(fmt.Sprintf("POST not signed: %v", parsedJws))
+		return nil, nil, errors.New("POST not signed")
+	}
+	// TODO: Look up key in registrations.
+	key := parsedJws.Signatures[0].Header.JsonWebKey
+	payload, err := parsedJws.Verify(key)
+	if err != nil {
+		wfe.log.Debug(string(body))
+		wfe.log.Debug(fmt.Sprintf("JWS verification error: %v", err))
+		return nil, nil, err
 	}
 
 	// TODO Return JWS body
-	return []byte(jws.Payload), jws.Header.Key, nil
+	return []byte(payload), key, nil
```

Comment:
I think the TODO is obsolete since we are already returning the body. `git blame` just shows me the monolithic original check in.